sqlalchemy-stubs/util/compat.pyi

```diff
@@ -216,3 +216,10 @@ else:
         def tzname(self, dt: Optional[datetime]) -> str: ...
         def dst(self, dt: Optional[datetime]) -> None: ...
         def fromutc(self, dt: datetime) -> datetime: ...
+
+if sys.version_info >= (3, 8):
+    from _typeshed import _SupportsIndex as SupportsIndex  # noqa
```

Comment:
Since this isn't exposed in the actual compat file, this should be called as `_SupportsIndex`